*** a/src/yaml_private.h	2014-01-13 01:24:55.000000000 +0100
--- b/src/yaml_private.h	2015-04-22 14:29:15.124511850 +0200
***************
*** 9,17 ****
  #include <limits.h>
  #include <stddef.h>
  
- #ifndef _MSC_VER
  #include <stdint.h>
! #else
  #ifdef _WIN64
  #define PTRDIFF_MAX _I64_MAX
  #else
--- 9,16 ----
  #include <limits.h>
  #include <stddef.h>
  
  #include <stdint.h>
! #ifdef _MSC_VER
  #ifdef _WIN64
  #define PTRDIFF_MAX _I64_MAX
  #else
*** a/src/Makefile.am	2015-04-22 14:10:00.418785980 +0200
--- b/src/Makefile.am	2015-04-23 09:14:49.366662367 +0200
***************
*** 1,4 ****
! AM_CPPFLAGS = -I$(top_srcdir)/include
  lib_LTLIBRARIES = libyaml.la
  libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
  libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
--- 1,4 ----
! AM_CPPFLAGS = -I$(top_srcdir)/include -D_INCLUDE_STDC__SOURCE_199901 -D__STDC_CONSTANT_MACROS
  lib_LTLIBRARIES = libyaml.la
  libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
  libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
*** a/src/Makefile.in	2014-01-13 01:25:09.000000000 +0100
--- b/src/Makefile.in	2015-04-23 09:16:06.787046266 +0200
***************
*** 216,222 ****
  top_build_prefix = @top_build_prefix@
  top_builddir = @top_builddir@
  top_srcdir = @top_srcdir@
! AM_CPPFLAGS = -I$(top_srcdir)/include
  lib_LTLIBRARIES = libyaml.la
  libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
  libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
--- 216,222 ----
  top_build_prefix = @top_build_prefix@
  top_builddir = @top_builddir@
  top_srcdir = @top_srcdir@
! AM_CPPFLAGS = -I$(top_srcdir)/include -D_INCLUDE_STDC__SOURCE_199901 -D__STDC_CONSTANT_MACROS
  lib_LTLIBRARIES = libyaml.la
  libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
  libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
