From 0fb7d90d3d85d0b6ca7f7b1d3969b17d2cfec274 Mon Sep 17 00:00:00 2001
From: Kristian Amlie <kristian.amlie@cfengine.com>
Date: Mon, 18 Jan 2016 15:18:56 +0100
Subject: [PATCH] Don't use "grep -q", "-q" is not POSIX, and fails on
 Solaris.

---
 util/domd |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/util/domd b/util/domd
index 6a628c7..95bb1b0 100755
--- a/util/domd
+++ b/util/domd
@@ -14,8 +14,8 @@ if [ "$MAKEDEPEND" = "" ]; then MAKEDEPEND=makedepend; fi
 cp Makefile Makefile.save
 # fake the presence of Kerberos
 touch $TOP/krb5.h
-if ${MAKEDEPEND} --version 2>&1 | grep -q "clang" ||
-   echo $MAKEDEPEND | grep -q "gcc"; then
+if ${MAKEDEPEND} --version 2>&1 | grep "clang" > /dev/null ||
+   echo $MAKEDEPEND | grep "gcc" > /dev/null; then
     args=""
     while [ $# -gt 0 ]; do
 	if [ "$1" != "--" ]; then args="$args $1"; fi
-- 
1.7.9.5

