From 8c827308e080353a7cda29ef22e65b6a30c4c4f1 Mon Sep 17 00:00:00 2001
From: Mikhail Gusarov <mikhail.gusarov@cfengine.com>
Date: Fri, 9 Sep 2011 16:21:03 +0200
Subject: [PATCH 2/2] Add mingw64 target

---
 Configure       |    4 +++-
 Makefile        |    2 +-
 Makefile.shared |    8 ++++++--
 3 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/Configure b/Configure
index ec87f24..9d6bba2 100755
--- a/Configure
+++ b/Configure
@@ -492,6 +492,8 @@ my %table=(
 # MinGW
 "mingw", "gcc:-mno-cygwin -DL_ENDIAN -fomit-frame-pointer -O3 -march=i486 -Wall -D_WIN32_WINNT=0x333:::MINGW32:-lwsock32 -lgdi32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${x86_coff_asm}:win32:cygwin-shared:-D_WINDLL -DOPENSSL_USE_APPLINK:-mno-cygwin -shared:.dll.a",
 
+"mingw64", "gcc:-DL_ENDIAN -O3 -Wall -D_WIN32_WINNT=0x333 -D_WIN32_DCOM -Wl,--export-all:::MINGW32:-lwsock32 -lgdi32:BN_LLONG ${x86_gcc_des} ${x86_gcc_opts} EXPORT_VAR_AS_FN:${x86_coff_asm}:win32:cygwin-shared:-D_WINDLL:-shared:.dll.a",
+
 # UWIN 
 "UWIN", "cc:-DTERMIOS -DL_ENDIAN -O -Wall:::UWIN::BN_LLONG ${x86_gcc_des} ${x86_gcc_opts}:${no_asm}:win32",
 
@@ -1084,7 +1086,7 @@ $IsMK1MF=1 if ($target eq "mingw" && $^O ne "cygwin" && !is_msys());
 
 $no_shared = 0 if ($fipsdso && !$IsMK1MF);
 
-$exe_ext=".exe" if ($target eq "Cygwin" || $target eq "DJGPP" || $target eq "mingw");
+$exe_ext=".exe" if ($target eq "Cygwin" || $target eq "DJGPP" || $target eq "mingw" || $target eq "mingw64");
 $exe_ext=".nlm" if ($target =~ /netware/);
 $exe_ext=".pm"  if ($target =~ /vos/);
 if ($openssldir eq "" and $prefix eq "")
diff --git a/Makefile b/Makefile
index 1b4cdc1..6556e5b 100644
--- a/Makefile
+++ b/Makefile
@@ -691,7 +691,7 @@ install_docs:
 	@pod2man="`cd ./util; ./pod2mantest $(PERL)`"; \
 	here="`pwd`"; \
 	filecase=; \
-	if [ "$(PLATFORM)" = "DJGPP" -o "$(PLATFORM)" = "Cygwin" -o "$(PLATFORM)" = "mingw" ]; then \
+	if [ "$(PLATFORM)" = "DJGPP" -o "$(PLATFORM)" = "Cygwin" -o "$(PLATFORM)" = "mingw" -o "$(PLATFORM)" = "mingw64" ]; then \
 		filecase=-i; \
 	fi; \
 	set -e; for i in doc/apps/*.pod; do \
diff --git a/Makefile.shared b/Makefile.shared
index 3183436..567020f 100644
--- a/Makefile.shared
+++ b/Makefile.shared
@@ -237,7 +237,9 @@ link_o.cygwin:
 	INHIBIT_SYMLINKS=yes; \
 	SHLIB=cyg$(LIBNAME); \
 	base=-Wl,--enable-auto-image-base; \
-	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
+	if expr $(PLATFORM) : 'mingw64' > /dev/null; then \
+		SHLIB=lib$(LIBNAME); base=; \
+	elif expr $(PLATFORM) : 'mingw' > /dev/null; then
 		SHLIB=$(LIBNAME)eay32; base=; \
 	fi; \
 	SHLIB_SUFFIX=.dll; \
@@ -252,7 +254,9 @@ link_a.cygwin:
 	INHIBIT_SYMLINKS=yes; \
 	SHLIB=cyg$(LIBNAME); \
 	base=-Wl,--enable-auto-image-base; \
-	if expr $(PLATFORM) : 'mingw' > /dev/null; then \
+	if expr $(PLATFORM) : 'mingw64' > /dev/null; then \
+		SHLIB=lib$(LIBNAME); base=; \
+	elif expr $(PLATFORM) : 'mingw' > /dev/null; then \
 		SHLIB=$(LIBNAME)eay32; \
 		base=;  [ $(LIBNAME) = "crypto" ] && base=-Wl,--image-base,0x63000000; \
 	fi; \
-- 
1.7.4.1

