*** a/Configure	2015-07-09 13:57:15.000000000 +0200
--- b/Configure	2015-10-14 07:03:34.080551550 +0200
***************
*** 1217,1223 ****
  my $unistd = $fields[$idx_unistd];
  my $thread_cflag = $fields[$idx_thread_cflag];
  my $sys_id = $fields[$idx_sys_id];
! my $lflags = $fields[$idx_lflags];
  my $bn_ops = $fields[$idx_bn_ops];
  my $cpuid_obj = $fields[$idx_cpuid_obj];
  my $bn_obj = $fields[$idx_bn_obj];
--- 1217,1223 ----
  my $unistd = $fields[$idx_unistd];
  my $thread_cflag = $fields[$idx_thread_cflag];
  my $sys_id = $fields[$idx_sys_id];
! my $lflags = join(" ", $fields[$idx_lflags], $ENV{'LDFLAGS'});
  my $bn_ops = $fields[$idx_bn_ops];
  my $cpuid_obj = $fields[$idx_cpuid_obj];
  my $bn_obj = $fields[$idx_bn_obj];
