diff -ruN curl-8.15.0/configure curl-8.15.0-modified/configure
--- curl-8.15.0/configure	2025-07-16 08:22:04.000000000 +0200
+++ curl-8.15.0-modified/configure	2025-07-21 16:35:32.470568787 +0200
@@ -41895,12 +41895,20 @@
   printf "%s\n" "#define HAVE_GETTIMEOFDAY 1" >>confdefs.h
 
 fi
-ac_fn_c_check_func "$LINENO" "if_nametoindex" "ac_cv_func_if_nametoindex"
-if test "x$ac_cv_func_if_nametoindex" = xyes
-then :
-  printf "%s\n" "#define HAVE_IF_NAMETOINDEX 1" >>confdefs.h
 
-fi
+# Below causes issues on Windows Build:
+#
+# 15:44:06 url.c: In function 'zonefrom_url':
+# 15:44:06 url.c:1802:18: error: implicit declaration of function 'if_nametoindex' [-Werror=implicit-function-declaration]
+# 15:44:06        scopeidx = if_nametoindex(zoneid);
+# 15:44:06                   ^
+
+# ac_fn_c_check_func "$LINENO" "if_nametoindex" "ac_cv_func_if_nametoindex"
+# if test "x$ac_cv_func_if_nametoindex" = xyes
+# then :
+#   printf "%s\n" "#define HAVE_IF_NAMETOINDEX 1" >>confdefs.h
+#
+# fi
 ac_fn_c_check_func "$LINENO" "mach_absolute_time" "ac_cv_func_mach_absolute_time"
 if test "x$ac_cv_func_mach_absolute_time" = xyes
 then :
@@ -48385,4 +48393,3 @@
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: ${curl_buildinfo}" >&5
 printf "%s\n" "$as_me: ${curl_buildinfo}" >&6;}
 fi
-
