diff -ruN curl-8.16.0/lib/curl_mem_undef.h curl-8.16.0-modified/lib/curl_mem_undef.h
--- curl-8.16.0/lib/curl_mem_undef.h    2025-09-10 07:43:43.000000000 +0200
+++ curl-8.16.0-modified/lib/curl_mem_undef.h   2025-09-24 16:51:34.546034324 +0200
@@ -44,6 +44,8 @@
 #undef socketpair
 #endif

+#ifdef CURLDEBUG
+
 #undef fopen
 #ifdef CURL_FOPEN
 #define fopen(fname, mode) CURL_FOPEN(fname, mode)
@@ -51,5 +53,7 @@
 #undef fdopen
 #undef fclose

+#endif /* CURLDEBUG */
+
 #undef HEADER_CURL_MEMORY_H
 #undef HEADER_CURL_MEMDEBUG_H
