--- configure.old
+++ configure
@@ -18114,6 +18114,10 @@ printf %s "checking for avx512 supports in compiler... " >&6; }
     #include <immintrin.h>
       int main(void) {
         __m512i mask = _mm512_set1_epi32(0x1);
+        __builtin_cpu_supports("avx512dq");
+        __builtin_cpu_supports("avx512cd");
+        __builtin_cpu_supports("avx512bw");
+        __builtin_cpu_supports("avx512vl");
         char out[32];
         _mm512_storeu_si512(out, _mm512_shuffle_epi8(mask, mask));
         return 0;
@@ -18152,6 +18156,7 @@ printf %s "checking for avx512 vbmi supports in compiler... " >&6; }
 
     #include <immintrin.h>
       int main(void) {
+        __builtin_cpu_supports("avx512vbmi");
         __m512i mask = _mm512_set1_epi32(0x1);
         char out[32];
         _mm512_storeu_si512(out, _mm512_permutexvar_epi8(mask, mask));
