diff --git a/ext/openssl/xp_ssl.c b/ext/openssl/xp_ssl.c
index ce23fb29..a200037e 100644
--- a/ext/openssl/xp_ssl.c
+++ b/ext/openssl/xp_ssl.c
@@ -1639,10 +1639,8 @@ int php_openssl_setup_crypto(php_stream *stream,
 
 	ssl_ctx_options &= ~SSL_OP_DONT_INSERT_EMPTY_FRAGMENTS;
 
-#ifdef SSL_OP_IGNORE_UNEXPECTED_EOF
 	/* Only for OpenSSL 3+ to keep OpenSSL 1.1.1 behavior */
 	ssl_ctx_options |= SSL_OP_IGNORE_UNEXPECTED_EOF;
-#endif
 
 	if (!GET_VER_OPT("disable_compression") || zend_is_true(val)) {
 		ssl_ctx_options |= SSL_OP_NO_COMPRESSION;
