*** a/libraries/liblmdb/mdb.c
--- b/libraries/liblmdb/mdb.c
***************
*** 2550,2557 **** mdb_txn_renew0(MDB_txn *txn)
  		}
  	} else {
  		if (ti) {
! 			mdb_mutex_t wmutex = MDB_MUTEX(env, w);
! 			if (LOCK_MUTEX(rc, env, wmutex))
  				return rc;
  
  			txn->mt_txnid = ti->mti_txnid;
--- 2550,2556 ----
  		}
  	} else {
  		if (ti) {
! 			if (LOCK_MUTEX(rc, env, MDB_MUTEX(env, w)))
  				return rc;
  
  			txn->mt_txnid = ti->mti_txnid;
***************
*** 9415,9421 **** static int mdb_reader_check0(MDB_env *env, int rlocked, int *dead)
  	mdb_mutex_t rmutex = rlocked ? NULL : MDB_MUTEX(env, r);
  	unsigned int i, j, rdrs;
  	MDB_reader *mr;
! 	pid_t *pids, pid;
  	int rc = MDB_SUCCESS, count = 0;
  
  	rdrs = env->me_txns->mti_numreaders;
--- 9414,9420 ----
  	mdb_mutex_t rmutex = rlocked ? NULL : MDB_MUTEX(env, r);
  	unsigned int i, j, rdrs;
  	MDB_reader *mr;
! 	MDB_PID_T *pids, pid;
  	int rc = MDB_SUCCESS, count = 0;
  
  	rdrs = env->me_txns->mti_numreaders;
