From e98434e9f769a64340666969e2bd9e42f8ad3845 Mon Sep 17 00:00:00 2001
From: Craig Comstock <craig@unreasonablefarm.org>
Date: Wed, 23 Nov 2022 14:14:46 -0600
Subject: [PATCH 13/15] enable debug logging always

---
 libraries/liblmdb/mdb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libraries/liblmdb/mdb.c b/libraries/liblmdb/mdb.c
index 27eb7a3..7f73f7c 100644
--- a/libraries/liblmdb/mdb.c
+++ b/libraries/liblmdb/mdb.c
@@ -476,6 +476,7 @@ typedef MDB_ID	txnid_t;
 /** @defgroup debug	Debug Macros
  *	@{
  */
+#define MDB_DEBUG 1
 #ifndef MDB_DEBUG
 	/**	Enable debug output.  Needs variable argument macros (a C99 feature).
 	 *	Set this to 1 for copious tracing. Set to 2 to add dumps of all IDLs
@@ -485,7 +486,7 @@ typedef MDB_ID	txnid_t;
 #endif
 
 #if MDB_DEBUG
-static int mdb_debug;
+static int mdb_debug = 1;
 static txnid_t mdb_debug_start;
 
 	/**	Print a debug message with printf formatting.
-- 
2.38.1

