diff --git a/lib/cmpbuf.c b/lib/cmpbuf.c
index aa79edb..bf0ee51 100644
--- a/lib/cmpbuf.c
+++ b/lib/cmpbuf.c
@@ -32,6 +32,10 @@
 # define SSIZE_MAX TYPE_MAXIMUM (ssize_t)
 #endif
 
+#ifndef SA_RESTART
+# define SA_RESTART 0
+#endif
+
 #undef MIN
 #define MIN(a, b) ((a) <= (b) ? (a) : (b))
 
diff --git a/lib/tzset.c b/lib/tzset.c
index 88fc047..fdbd688 100644
--- a/lib/tzset.c
+++ b/lib/tzset.c
@@ -18,6 +18,8 @@
 /* written by Jim Meyering */
 
 #include <config.h>
+#include <stdlib.h>
+#include <string.h>
 
 /* Specification.  */
 #include <time.h>
diff --git a/src/system.h b/src/system.h
index 469522d..a13c99f 100644
--- a/src/system.h
+++ b/src/system.h
@@ -65,7 +65,7 @@
 #include <string.h>
 #if ! HAVE_STRCASECOLL
 # if HAVE_STRICOLL || defined stricoll
-#  define strcasecoll(a, b) stricoll (a, b)
+#  define strcasecoll(a, b) _stricoll (a, b)
 # else
 #  define strcasecoll(a, b) strcasecmp (a, b) /* best we can do */
 # endif
diff --git a/src/util.c b/src/util.c
index b25dbe1..7705a12 100644
--- a/src/util.c
+++ b/src/util.c
@@ -273,12 +273,14 @@ process_signals (void)
       /* SIGTSTP is special, since the application can receive that signal
          more than once.  In this case, don't set the signal handler to the
          default.  Instead, just raise the uncatchable SIGSTOP.  */
+/* don't try and handle SIGTSTP on windows, not available
       if (stops)
         {
           stop_signal_count = stops - 1;
           sig = SIGSTOP;
         }
       else
+*/
 	{
 	  sig = interrupt_signal;
 	  xsignal (sig, SIG_DFL);
@@ -307,7 +309,15 @@ static int const sig[] =
 #ifdef SIGALRM
     SIGALRM,
 #endif
-    SIGHUP, SIGINT, SIGPIPE,
+#ifdef SIGHUP
+    SIGHUP,
+#endif
+#ifdef SIGINT
+    SIGINT,
+#endif
+#ifdef SIGPIPE
+    SIGPIPE,
+#endif
 #ifdef SIGQUIT
     SIGQUIT,
 #endif
