*** ./src/Makefile.shlib	Tue Jun 18 12:45:26 2013
--- ./src/Makefile.shlib	Tue Jun 18 12:50:09 2013
***************
*** 116,122 ****
  
  ifeq ($(PORTNAME), aix)
    ifdef SO_MAJOR_VERSION
-     shlib		= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
    endif
    haslibarule   = yes
    exports_file		= lib$(NAME).exp
--- 116,123 ----
  
  ifeq ($(PORTNAME), aix)
    ifdef SO_MAJOR_VERSION
+     shlib				= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
+     shlib_major		= lib$(NAME)$(DLSUFFIX).$(SO_MAJOR_VERSION)
    endif
    haslibarule   = yes
    exports_file		= lib$(NAME).exp
***************
*** 473,481 ****
  ifeq ($(enable_shared), yes)
  install-lib-shared: $(shlib) installdirs-lib
  ifdef soname
- # we don't install $(shlib) on AIX
- # (see http://archives.postgresql.org/message-id/52EF20B2E3209443BC37736D00C3C1380A6E79FE@EXADV1.host.magwien.gv.at)
- ifneq ($(PORTNAME), aix)
  	$(INSTALL_SHLIB) $< '$(DESTDIR)$(libdir)/$(shlib)'
  ifneq ($(PORTNAME), cygwin)
  ifneq ($(PORTNAME), win32)
--- 474,479 ----
***************
*** 491,497 ****
  endif
  endif # not win32
  endif # not cygwin
- endif # not aix
  else # no soname
  	$(INSTALL_SHLIB) $< '$(DESTDIR)$(pkglibdir)/$(shlib)'
  endif
--- 489,494 ----
