--- pg_hba.conf.sample	2019-11-11 16:03:10.000000000 -0600
+++ pg_hba.conf.socket-only	2020-01-21 13:22:44.119994084 -0600
@@ -79,11 +79,11 @@
 @remove-line-for-nolocal@# "local" is for Unix domain socket connections only
 @remove-line-for-nolocal@local   all             all                                     @authmethodlocal@
 # IPv4 local connections:
-host    all             all             127.0.0.1/32            @authmethodhost@
+#host    all             all             127.0.0.1/32            @authmethodhost@
 # IPv6 local connections:
-host    all             all             ::1/128                 @authmethodhost@
+#host    all             all             ::1/128                 @authmethodhost@
 # Allow replication connections from localhost, by a user with the
 # replication privilege.
 @remove-line-for-nolocal@local   replication     all                                     @authmethodlocal@
-host    replication     all             127.0.0.1/32            @authmethodhost@
-host    replication     all             ::1/128                 @authmethodhost@
+#host    replication     all             127.0.0.1/32            @authmethodhost@
+#host    replication     all             ::1/128                 @authmethodhost@
