diff -rupN redis-2.8.2.old/deps/hiredis/Makefile redis-2.8.2.new/deps/hiredis/Makefile
--- redis-2.8.2.old/deps/hiredis/Makefile	2013-12-02 16:07:46.000000000 +0100
+++ redis-2.8.2.new/deps/hiredis/Makefile	2013-12-13 22:52:11.206403701 +0100
@@ -15,8 +15,8 @@ CC:=$(shell sh -c 'type $(CC) >/dev/null
 OPTIMIZATION?=-O3
 WARNINGS=-Wall -W -Wstrict-prototypes -Wwrite-strings
 DEBUG?= -g -ggdb
-REAL_CFLAGS=$(OPTIMIZATION) -fPIC $(CFLAGS) $(WARNINGS) $(DEBUG) $(ARCH)
-REAL_LDFLAGS=$(LDFLAGS) $(ARCH)
+REAL_CFLAGS=$(OPTIMIZATION) -fPIC $(CFLAGS) $(WARNINGS) $(DEBUG)
+REAL_LDFLAGS=$(LDFLAGS)
 
 DYLIBSUFFIX=so
 STLIBSUFFIX=a
