diff --git a/lib/saslutil.c b/lib/saslutil.c
index 6174124..d7b325f 100644
--- a/lib/saslutil.c
+++ b/lib/saslutil.c
@@ -80,9 +80,11 @@
  * sasl_erasebuffer
  */
 
-#ifdef sun
-/* gotta define gethostname ourselves on suns */
-extern int gethostname(char *, int);
+#ifndef HAVE_UNISTD_H /*solaris 11 defines its own gethostname with different signature */
+    #ifdef sun
+    /* gotta define gethostname ourselves on suns */
+    extern int gethostname(char *, int);
+    #endif
 #endif
 
 char *encode_table;
