From 2107d1dbd442f39eba3b02fb165020e13d18cb1d Mon Sep 17 00:00:00 2001
From: Mikhail Gusarov <mikhail.gusarov@cfengine.com>
Date: Thu, 15 Sep 2011 16:08:46 +0200
Subject: [PATCH] Prefix windres with cross-compiler name.

---
 libraries/liblutil/Makefile.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libraries/liblutil/Makefile.in b/libraries/liblutil/Makefile.in
index afafab4..29c8a6a 100644
--- a/libraries/liblutil/Makefile.in
+++ b/libraries/liblutil/Makefile.in
@@ -54,7 +54,7 @@ slapdmsg.bin: FORCE
 	@if [ ! -f $@ ]; then cp $(srcdir)/$@ .; fi
 
 slapdmsg.res: slapdmsg.rc slapdmsg.bin
-	windres $< -O coff -o $@
+	@host@-windres $< -O coff -o $@
 
 clean-local:
 	$(RM) *.res
-- 
1.7.5.4

