--- configure	2018-03-22 16:44:27.000000000 +0100
+++ configure.new	2019-05-13 18:28:03.002377386 +0200
@@ -6711,7 +6711,7 @@
 # whether `pass_all' will *always* work, you probably want this one.
 
 case $host_os in
-aix4* | aix5*)
+aix4* | aix5* | aix7*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
@@ -9059,7 +9059,7 @@
 
     # See if GNU ld supports shared libraries.
     case $host_os in
-    aix3* | aix4* | aix5*)
+    aix3* | aix4* | aix5* | aix7*)
       # On AIX/PPC, the GNU linker is very broken
       if test "$host_cpu" != ia64; then
 	ld_shlibs=no
@@ -9227,7 +9227,7 @@
       fi
       ;;
 
-    aix4* | aix5*)
+    aix4* | aix5* | aix7*)
       if test "$host_cpu" = ia64; then
 	# On IA64, the linker does run time linking by default, so we don't
 	# have to do anything special.
@@ -9247,7 +9247,7 @@
 	# Test if we are trying to use run time linking or normal
 	# AIX style linking. If -brtl is somewhere in LDFLAGS, we
 	# need to do runtime linking.
-	case $host_os in aix4.[23]|aix4.[23].*|aix5*)
+	case $host_os in aix4.[23]|aix4.[23].*|aix5*|aix7*)
 	  for ld_flag in $LDFLAGS; do
   	  if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
   	    aix_use_runtimelinking=yes
@@ -9922,7 +9922,7 @@
   soname_spec='${libname}${release}${shared_ext}$major'
   ;;
 
-aix4* | aix5*)
+aix4* | aix5* | aix7*)
   version_type=linux
   need_lib_prefix=no
   need_version=no
@@ -11010,7 +11010,7 @@
   fi
   ;;
 
-aix4* | aix5*)
+aix4* | aix5* | aix7*)
   if test "$host_cpu" != ia64 && test "$aix_use_runtimelinking" = no ; then
     test "$enable_shared" = yes && enable_static=no
   fi
