--- aclocal.m4.orig	2011-11-28 11:36:55.000000000 +0100
+++ aclocal.m4	2011-11-28 11:38:33.000000000 +0100
@@ -2699,7 +2699,7 @@
 mingw* | pw32*)
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump'.
-  lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
+  lt_cv_deplibs_check_method='file_magic file format (pei*-i386(.*architecture: i386)?|pe-x86-64)'
   lt_cv_file_magic_cmd='$OBJDUMP -f'
   ;;
 
--- build/ltmain.sh.orig	2011-11-28 11:38:17.000000000 +0100
+++ build/ltmain.sh	2011-11-28 11:38:43.000000000 +0100
@@ -209,7 +209,7 @@
     ;;
   *ar\ archive*) # could be an import, or static
     if eval $OBJDUMP -f $1 | $SED -e '10q' 2>/dev/null | \
-      $EGREP -e 'file format pe-i386(.*architecture: i386)?' >/dev/null ; then
+      $EGREP -e 'file format (pe-i386(.*architecture: i386)?|pe-x86-64)' >/dev/null ; then
       win32_nmres=`eval $NM -f posix -A $1 | \
 	$SED -n -e '1,100{/ I /{s,.*,import,;p;q;};}'`
       case $win32_nmres in
--- configure.orig	2011-11-28 11:41:29.000000000 +0100
+++ configure	2011-11-28 11:41:42.000000000 +0100
@@ -6290,7 +6290,7 @@
 mingw* | pw32*)
   # Base MSYS/MinGW do not provide the 'file' command needed by
   # func_win32_libid shell function, so use a weaker test based on 'objdump'.
-  lt_cv_deplibs_check_method='file_magic file format pei*-i386(.*architecture: i386)?'
+  lt_cv_deplibs_check_method='file_magic file format (pei*-i386(.*architecture: i386)?|pe-x86-64)'
   lt_cv_file_magic_cmd='$OBJDUMP -f'
   ;;
 
