--- include/ldap_pvt.h.orig	2021-06-09 13:25:02.701875533 +0200
+++ include/ldap_pvt.h	2021-06-09 21:50:23.356663177 +0200
@@ -185,7 +185,10 @@
 #define LDAP_IPADDRLEN	sizeof("IP=255.255.255.255:65336")
 #endif
 
+#ifndef _SOCKADDR_UNION_DEFINED
 union Sockaddr;
+#define _SOCKADDR_UNION_DEFINED
+#endif
 
 LDAP_F (void)
 ldap_pvt_sockaddrstr LDAP_P((
--- include/ac/socket.h.orig	2021-06-09 21:08:51.197144446 +0200
+++ include/ac/socket.h	2021-06-09 21:50:10.931819292 +0200
@@ -234,6 +234,7 @@
 #define	LUTIL_GETPEEREID( s, uid, gid, bv )	lutil_getpeereid( s, uid, gid )
 #endif
 
+#ifndef _SOCKADDR_UNION_DEFINED
 typedef union Sockaddr {
 	struct sockaddr sa_addr;
 	struct sockaddr_in sa_in_addr;
@@ -245,6 +246,8 @@
 	struct sockaddr_un sa_un_addr;
 #endif
 } Sockaddr;
+#define _SOCKADDR_UNION_DEFINED
+#endif  /* _SOCKADDR_UNION_DEFINED */
 
 /* DNS RFC defines max host name as 255. New systems seem to use 1024 */
 #ifndef NI_MAXHOST
