From 194224ffb23fb3715709a4ad40c29af39ccbe4bf Mon Sep 17 00:00:00 2001
From: Mikhail Gusarov <mikhail.gusarov@cfengine.com>
Date: Thu, 8 Sep 2011 00:56:40 +0200
Subject: [PATCH] Fix struct timespec redifinition for MinGW

---
 config.h  |    1 +
 pthread.h |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/config.h b/config.h
index d6638df..ba6bf54 100644
--- a/config.h
+++ b/config.h
@@ -116,6 +116,7 @@
 
 #ifdef __MINGW32__
 #define HAVE_MODE_T
+#define HAVE_STRUCT_TIMESPEC
 #endif
 
 #ifdef __BORLANDC__
diff --git a/pthread.h b/pthread.h
index f3d2dac..2988235 100644
--- a/pthread.h
+++ b/pthread.h
@@ -247,6 +247,7 @@ enum {
 #  endif
 #  if defined(_UWIN) || defined(__MINGW32__)
 #    define HAVE_MODE_T
+#    define HAVE_STRUCT_TIMESPEC 1
 #  endif
 #endif
 
-- 
1.7.5.4

