From 2bdbd5ec549474ba09fe1399cfa7e15cc91aa9ca Mon Sep 17 00:00:00 2001
From: Mikhail Gusarov <mikhail.gusarov@cfengine.com>
Date: Mon, 22 Aug 2011 14:35:55 +0200
Subject: [PATCH] Compile out ifaceGetVlanID if no MacVTap is requested

---
 src/util/interface.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/src/util/interface.c b/src/util/interface.c
index eec437a..c567b06 100644
--- a/src/util/interface.c
+++ b/src/util/interface.c
@@ -348,6 +348,7 @@ ifaceGetIndex(bool reportError,
 
 #endif /* __linux__ */
 
+#if defined(HAVE_MACVTAP)
 #ifdef __linux__
 int
 ifaceGetVlanID(const char *vlanifname, int *vlanid) {
@@ -390,3 +391,4 @@ ifaceGetVlanID(const char *vlanifname ATTRIBUTE_UNUSED,
     return ENOSYS;
 }
 #endif /* __linux__ */
+#endif /* HAVE_MACVTAP */
-- 
1.7.4.1

