*** Makefile.in	Wed Apr 17 15:17:20 2013
--- Makefile.in	Wed Apr 17 15:12:59 2013
***************
*** 488,494 ****
  	  then \
  	    $(CC) -shared -Wl,-G,-h,libqdbm.so.$(LIBVER) -o $@ $(MYLIBOBJS) $(LIBLDFLAGS) ; \
  	  else \
! 	    $(CC) -shared -Wl,-soname,libqdbm.so.$(LIBVER) -o $@ $(MYLIBOBJS) $(LIBLDFLAGS) ; \
  	  fi
  
  
--- 488,494 ----
  	  then \
  	    $(CC) -shared -Wl,-G,-h,libqdbm.so.$(LIBVER) -o $@ $(MYLIBOBJS) $(LIBLDFLAGS) ; \
  	  else \
! 	    $(CC) -shared -Wl,-G,+O,libqdbm.so.$(LIBVER) -o $@ $(MYLIBOBJS) $(LIBLDFLAGS) ; \
  	  fi
  
  
