--- pcre_internal.h.orig	2012-03-13 16:19:53.966210786 +0100
+++ pcre_internal.h	2012-03-13 16:19:41.094210593 +0100
@@ -91,6 +91,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include <sys/int_types.h>
+
 /* When compiling a DLL for Windows, the exported symbols have to be declared
 using some MS magic. I found some useful information on this web page:
 http://msdn2.microsoft.com/en-us/library/y4h7bcy6(VS.80).aspx. According to the
